### PR TITLE
UI for recently sold products

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #8571 

### Description

This PR includes UI for displaying the recent products section in the product selection screen found in order creation.

To understand how this is calculated, please refer to this post: pdfdoF-2At-p2

If there are no recent products, the entire section will be hidden. Also, while searching for products, the recent section will be hidden and shown back again when the search query is cleared.

#### Recent products section displayed
1. Make sure to log into the store that has many orders paid for with some products.
2. Navigate to the Order creation screen (Orders -> click on + button)
3. Click on the "Add products" button
4. Make sure you see the recent products section

#### Recent products section is hidden when there are no recent products
1. Make sure to log into the store that has just simple payment orders where no products are added while creating the order.
2. Navigate to the Order creation screen (Orders -> click on + button)
3. Click on the "Add products" button
4. Make sure you do not see the recent products section.

#### Recent products section is hidden when searching for products
1. Make sure to log into the store that has many orders paid for with some products.
2. Navigate to the Order creation screen (Orders -> click on + button)
3. Click on the "Add products" button
4. Make sure you see the recent products section
5. Search for any products
6. Make sure the recent products section gets hidden
7. Clear the search query
8. Make sure the recent products section gets shown again.

### Images/gif

https://user-images.githubusercontent.com/1331230/229059392-021a5e78-cf36-47ec-8d2b-76d255e279b2.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt [hunk 1]
```diff
@@ -194,7 +226,7 @@ private fun PopularProductsList(
             )
         }
 
-        state.popularProducts.forEachIndexed { index, popularProduct ->
+        productsList.forEachIndexed { index, popularProduct ->
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt]:
```suggestion
        productsList.forEachIndexed { index, product ->
```
💡 It can be both popular and recent.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt [hunk 2]
```diff
@@ -173,16 +173,48 @@ private fun EmptyProductList(
...
+    displayProductsSection(
+        type = ProductType.RECENT,
+        state = state,
+        onProductClick = onProductClick
+    )
+}
+
+@Composable
+private fun displayProductsSection(
+    type: ProductType,
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt:199]: 💡 In general, we use `type` to get the header's text only. Why not pass the string instead of the type and remove `ProductType` enum to simplify the code?

**Review by samiuelson — COMMENTED** [review]
Good work, @AnirudhBhat! I've added a few code improvement ideas for your consideration.

❓ Other than that, there is an issue that I think is worth fixing - the recently sold items section shows duplicated products. See the screenshot below. In order to reproduce it just create a few orders containing the same product.


💡 It would be reasonable to create a unit test for this edge case as well.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt]: Agree https://github.com/woocommerce/woocommerce-android/pull/8690/commits/0c026c77964ee7ead3e829a2b0cbdca0bb6c8c68

**Comment by AnirudhBhat** [comment]
>The recently sold items section shows duplicated products

Good catch! I've fixed this and it should work fine now. I've added the corresponding unit test as well.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt (hunk 2 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/selector/ProductSelectorScreen.kt:199]: `type` is used to decide which product list to populate on the UI as well.

```
val (productsList, heading) = when (type) {
        ProductType.POPULAR -> Pair(
            state.popularProducts, stringResource(id = string.product_selector_popular_products_heading)
        )
        ProductType.RECENT -> Pair(
            state.recentProducts, stringResource(id = string.product_selector_recent_products_heading)
        )
    }
```

I feel like using string here is kinda risky and prone to errors. WDYT?

**Review by samiuelson — APPROVED** [review]
Looks good! Good work 🎉
